src/gql/ProgramDonorPublishedAnalysisByDateRange/resolvers/index.ts

```diff
@@ -17,12 +17,26 @@
  * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-import gql from "graphql-tag";
+import { Client } from "@elastic/elasticsearch";
+import { resolveWithProgramAuth } from "../../ProgramDonorSummary/resolvers";
+import programDonorPublishedAnalysisByDateRangeResolver from "./programDonorPublishedAnalysisByDateRange";
+import { GlobalGqlContext } from "gql/server";
+import { IResolvers } from "@graphql-tools/utils";
 
-const typeDefs = gql`
-  type Query {
-    ping: Boolean
-  }
-`;
+const createResolvers = async (
+  esClient: Client
+): Promise<IResolvers<unknown, GlobalGqlContext>> => {
+  return {
+    Query: {
+      programDonorPublishedAnalysisByDateRange: (...resolverArguments) =>
+        resolveWithProgramAuth(
+          programDonorPublishedAnalysisByDateRangeResolver(esClient)(
+            ...resolverArguments
+          ),
+          resolverArguments
+        ),
```

Comment:
At a glance, this block is really complicated
Nested functions, `resolverArguments` passed three times, a client in the body of a Query ... my first reaction is this could use some untangling

Reply:
I agree. Copying the code "as-is" as per ticket requirements.